src/components/ui/RelatedSearch.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useHawksearch } from 'components/StoreProvider';
+
+function RelatedSearch() {
+	const {
+		actor: hawkActor,
+		store: { searchResults },
+	} = useHawksearch();
+	const relatedFacet = searchResults && searchResults.Facets.find(facet => facet.FacetType === 'related');
+	console.log(relatedFacet);
+
+	function searchWithKeyword(keyword) {
+		hawkActor.setSearch({
+			Keyword: keyword,
+		});
+	}
+	return (
+		<>
+			{relatedFacet && (
+				<div className="hawk-related_search-container">
+					<span className="heading">Related Searches: </span>
+					{relatedFacet.Values.map((item, index) => {
+						console.log(index, relatedFacet.Values.length);
+						return (
+							<span
+								key={index}
+								onClick={() => searchWithKeyword(item.Value)}
+								className="related-searched-words"
+							>
+								{' '}
```

Comment:
This part is hard to understand - is there a chance to build it in a way that would be understandable without deeper look /

Reply:
Simplified the logic